url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/data/gathering/dataSource/ProquestGoogleFinanceNYSE.java

```diff
@@ -0,0 +1,313 @@
+						webize(companyname) +
+						"%22&submit=Search&date=RANGE&onDate=&beforeDate=&afterDate=&fromDate=" +
+						prosearchdconv(this.startdate) +
+						"&toDate=" +
+						prosearchdconv(this.enddate) +
+						"&pubtitle=&author=&FT=0&AT=news&revType=review&revPos=all&STYPE=all&sortby=REVERSE_CHRON";
+				Date firsttry = readIn(ticker,proquest,nostories);
+				while(!firsttry.equals(startdate)) {
+					String newpro = "http://proquest.umi.com.proxy.library.cornell.edu/pqdweb?DBId=5921&RQT=305&querySyntax=PQ&searchInterface=1&moreOptState=CLOSED&TS=1257203651&h_pubtitle=&h_pmid=&JSEnabled=1&SQ=%22" +
+					webize(companyname) +
+					"%22&submit=Search&date=RANGE&onDate=&beforeDate=&afterDate=&fromDate=" +
+					prosearchdconv(this.startdate) +
+					"&toDate=" +
+					prosearchdconv(firsttry) +
+					"&pubtitle=&author=&FT=0&AT=news&revType=review&revPos=all&STYPE=all&sortby=REVERSE_CHRON";
+					firsttry = readIn(ticker,newpro,nostories);
+				}
+				String companyid = lookupGooglecompanyid(currentstock);
+				GregorianCalendar cal = new GregorianCalendar();
+				cal.setTime(startdate);
+				String startmonth = csvreader.MonthtoGoogle(cal);
+				String startday = String.valueOf(cal.get(Calendar.DAY_OF_MONTH));
+				String startyear = String.valueOf(cal.get(Calendar.YEAR));
+				cal.setTime(enddate);
+				String endmonth = csvreader.MonthtoGoogle(cal);
+				String endday = String.valueOf(cal.get(Calendar.DAY_OF_MONTH));
+				String endyear = String.valueOf(cal.get(Calendar.YEAR));
+				String longurl = "http://www.google.com/finance/historical?cid=" + companyid + "&startdate=" 
+				+ startmonth + "+" + startday + "%2C+" + startyear + "&enddate=" + endmonth + "+" + endday + "%2C+"
+				+ endyear + "&output=csv";
+				URL googlecsv = new URL(longurl);
+				System.out.println(currentstock + ":" + longurl);
+				BufferedReader googlecsvread = new BufferedReader(new InputStreamReader(googlecsv.openStream()));
+				ticker.addData(csvreader.readIn(googlecsvread));
+				if (googlecsvread != null) {
+					googlecsvread.close();
+				}
+				addTicker(ticker);
+			}
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} finally {
+			if (s != null) {
+				s.close();
+			}
+		}
+	}
+	
+	public String lookupGooglecompanyid(String symbol) throws IOException {
+		try {
+			URL google = new URL("http://www.google.com/finance?q=NYSE:" + symbol);
+			BufferedReader in = new BufferedReader(new InputStreamReader(google.openStream()));
+			Scanner s = new Scanner(in);
+			s.useDelimiter("[\n\r]+");
+			String companyid = null;
+			while(s.hasNext()) {
+				String next = s.next();
+				if(next.contains("var _companyId")) {
+					String extractor = next.substring(17);
+					companyid = extractor.substring(0,(extractor.indexOf(';')));
+					break;
+				}
+			}
+			if (s != null) {
+				s.close();
+			}
+			if (in != null) {
+				in.close();
+			}
+			return companyid;
+		}
+		catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)